exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/common/ChainedHashTable.java

```diff
@@ -208,7 +216,10 @@ public HashTable createAndSetupHashTable(TypedFieldId[] outKeyFieldIds) throws C
       i++;
     }
 
-
+    // Only in case of an equi-join: Generate a special method to check if both the new key and the existing key (in this HT bucket) are nulls
```

Comment:
When you say equi-join, do you mean inner join ?  The join operators don't handle non-equijoins currently.

Reply:
The 'equi-join' term was just following the 'Comparator.EQUALS'; and is correct as we only support these now.
   The point here is that for *all joins*, nulls are equal on build/insert, but not equal on probe. This works the same for outer joins, as an unmatched probe on a single hash-table key is the same as un-matching multiple such keys.  Maybe I should add this to the comment to clarify.